Move threadpool to extension

kirklandsign opened the pull request:
Summary:
1. Replace `executorch/backends/xnnpack/threadpool` with `executorch/extension/threadpool`
2. `hg mv xplat/executorch/backends/xnnpack/threadpool xplat/executorch/extension` and `hg mv fbcode/executorch/backends/xnnpack/threadpool fbcode/executorch/extension`

Differential Revision: D61938189

larryliu0820 reviewed (approved):
@kimishpatel should take a look

kirklandsign committed:
Move threadpool to extension (#4954)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4954

1. Replace `executorch/backends/xnnpack/threadpool` with `executorch/extension/threadpool` Need to exclude `whatsapp/`
2. `hg mv xplat/executorch/backends/xnnpack/threadpool xplat/executorch/extension` and `hg mv fbcode/executorch/backends/xnnpack/threadpool fbcode/executorch/extension`

Reviewed By: larryliu0820

Differential Revision: D61938189
